applications/main/bad_usb/bad_usb_script.c

```diff
@@ -204,16 +207,29 @@ static bool ducky_altstring(const char* param) {
     return state;
 }
 
-static bool ducky_string(const char* param) {
+static bool ducky_string(BadUsbScript* bad_usb, const char* param) {
     uint32_t i = 0;
+    uint32_t timing = 0;
+    
+    if(bad_usb->stringdelay == 0) {
+        timing = 0;
+    } else if(bad_usb->stringdelay > 1000) {
+        timing = bad_usb->stringdelay / 2 / 10;
+    } else {
+        timing = bad_usb->stringdelay / 2;
+    }
+
     while(param[i] != '\0') {
         uint16_t keycode = HID_ASCII_TO_KEY(param[i]);
         if(keycode != HID_KEYBOARD_NONE) {
+            furi_delay_ms(timing);
```

Comment:
`furi_delay_ms(0)` causes task yield and may lead to performance lowering, it's better to check timing value before calling delay